### PR TITLE
Fix io_uring failure

### PR DESCRIPTION
**Pull request by chunfuwen**
queue_size in driver is not supported

### CODE REVIEW

**Comment by chunfuwen** [comment]
Failed log:

2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|     run_test_case(test, params, env)
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|   File "/var/lib/avocado/data/avocado-vt/virttest/test-providers.d/downloads/io-github-autotest-libvirt/libvirt/tests/src/virtual_disks/virtual_disks_io_tuning.py", line 87, in test_coldplug_io_uring_normal_start
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|     virsh.attach_device(vm_name, device_obj.xml, flagstr="--config",
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|   File "/var/ci/libvirt-ci/runtest/avocado-vt/avocado-vt/virttest/virsh.py", line 1782, in attach_device
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|     return _adu_device("attach-device", domainarg=domainarg, filearg=filearg,
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|   File "/var/ci/libvirt-ci/runtest/avocado-vt/avocado-vt/virttest/virsh.py", line 1765, in _adu_device
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|     return command(cmd, **dargs)
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|   File "/var/ci/libvirt-ci/runtest/avocado-vt/avocado-vt/virttest/virsh.py", line 820, in command
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|     ret = process.run(cmd, timeout=timeout, verbose=debug,
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|   File "/var/ci/libvirt-ci/runtest/avocado-vt/avocado/avocado/utils/process.py", line 1093, in run
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR|     raise CmdError(cmd, sp.result)
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR| avocado.utils.process.CmdError: Command '/bin/virsh attach-device avocado-vt-vm1 /tmp/xml_utils_temp_s7pygv32.xml --config' failed.
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR| stdout: b'\n'
2024-01-18 14:46:08,764 stacktrace       L0045 ERROR| stderr: b'error: Failed to attach device from /tmp/xml_utils_temp_s7pygv32.xml\nerror: unsupported configuration: queue_size attribute in disk driver

**Review by cliping — APPROVED** [review]
LGTM

**Comment by chunfuwen** [comment]
Command 'virsh list' finished with 0 after 0.034083162s
DATA (filename=output.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stdout.expected) => NOT FOUND (data sources: variant, test, file)
DATA (filename=stderr.expected) => NOT FOUND (data sources: variant, test, file)
PASS 1-type_specific.io-github-autotest-libvirt.virtual_disks.io_tuning.hotplug.io_uring.normal_start

File: libvirt/tests/cfg/virtual_disks/virtual_disks_io_tuning.cfg [hunk 1]
```diff
@@ -7,7 +7,7 @@
             func_supported_since_libvirt_ver = (9, 3, 0)
             variants test_scenario:
                 - normal_start:
-                    driver_attribute = {'name': "qemu", 'type': "qcow2", 'queue_size': '1024', 'io': 'io_uring'}
+                    driver_attribute = {'name': "qemu", 'type': "qcow2", 'io': 'io_uring'}
```
Comment (smitterl) [inline libvirt/tests/cfg/virtual_disks/virtual_disks_io_tuning.cfg:10]: I can start a VM successfully with
```
<driver name='qemu' type='qcow2' queue_size='1024'/>
```
and `libvirt-10.0.0`

Comment (chunfuwen) [reply libvirt/tests/cfg/virtual_disks/virtual_disks_io_tuning.cfg:10]: you miss 'io': 'io_uring' parameter, if adding this parameter ,it will raise issue

Comment (smitterl) [reply libvirt/tests/cfg/virtual_disks/virtual_disks_io_tuning.cfg:10]: @chunfuwen I forgot to add that in my comment you are right. But in any case it still works with io_uring:

The VM starts for me with
```
 -blockdev {"driver":"file","filename":"/var/lib/avocado/data/avocado-vt/images/jeos-27-s390x.qcow2","aio":"io_uring","node-name":"libvirt-1-storage","auto-read-only":true,"discard":"unmap"} -blockdev {"node-name":"libvirt-1-format","read-only":false,"driver":"qcow2","file":"libvirt-1-storage","backing":null} -device {"driver":"virtio-blk-ccw","queue-size":1024,"devno":"fe.0.0000","drive":"libvirt-1-format","id":"virtio-disk0","bootindex":1} ```

I had a look at the original issue, the error says "unsupported configuration: queue_size attribute in disk driver is only supported for virtio bus". Actually, the test case tries to use 'scsi':

```
<disk type="file" device="disk"><target dev="sdb" bus="scsi" /><driver name="qemu" type="qcow2" queue_size="1024" io="io_uring" /><source file="/var/lib/libvirt/images/io_uring.qcow2" /></disk>```

And yes, with that I can see that error message.

**Review by smitterl — CHANGES_REQUESTED** [review]
Please check my comment.

File: libvirt/tests/cfg/virtual_disks/virtual_disks_io_tuning.cfg (hunk 1 above)
Comment (chunfuwen) [reply libvirt/tests/cfg/virtual_disks/virtual_disks_io_tuning.cfg:10]: queue_size is supported only virto, which is latest change. Previously It is support on both virto and scsi. That is why we have this case. So this fix is to remove queue_size from scsi. 
But more further task is to add virtio case related to io_uring

Comment (smitterl) [reply libvirt/tests/cfg/virtual_disks/virtual_disks_io_tuning.cfg:10]: @chunfuwen I understand it now. Thank you for explaining.

**Review by smitterl — APPROVED** [review]
LGTM

**Review by dzhengfy — APPROVED** [review]
LGTM.
A reminder is to rename this case to reflect the scsi bus later.

**Pull request merged**